compiler/shared/main/scala/mlscript/compiler/codegen/CppCodeGen.scala

```diff
@@ -0,0 +1,242 @@
+      codegen(body, storeInto)(using decls, stmts2)
+    case Node.LetMethodCall(names, cls, method, args, body) =>
+      val call = mlsMethodCall(cls, method.str |> mapName, args.map(toExpr))
+      val stmts2 = stmts ++ Ls(Stmt.AutoBind(names.map(mapName), call))
+      codegen(body, storeInto)(using decls, stmts2)
+    // Use method calls instead of apply
+    // 
+    // case Node.LetApply(names, fn, args, body) if fn.str == "builtin" =>
+    //   val stmts2 = stmts ++ codegenBuiltin(names, args.head.toString.replace("\"", ""), args.tail)
+    //   codegen(body, storeInto)(using decls, stmts2)
+    // case Node.LetApply(names, fn, args, body) =>
+    //   val call = mlsCall(fn.str |> mapName, args.map(toExpr))
+    //   val stmts2 = stmts ++ Ls(Stmt.AutoBind(names.map(mapName), call))
+    //   codegen(body, storeInto)(using decls, stmts2)
+    case Node.LetCall(names, defn, args, _, body) =>
+      val call = Expr.Call(Expr.Var(defn.name |> mapName), args.map(toExpr))
+      val stmts2 = stmts ++ Ls(Stmt.AutoBind(names.map(mapName), call))
+      codegen(body, storeInto)(using decls, stmts2)
+    case Node.Case(scrut, cases, default) =>
+      codegenCaseWithIfs(scrut, cases, default, storeInto)
+    
+  private def codegenDefn(using ctx: Ctx)(defn: Defn): (Def, Decl) = defn match
+    case Defn(id, name, params, resultNum, body, _, _) =>
+      val decls = Ls(mlsRetValueDecl)
+      val stmts = Ls.empty[Stmt]
+      val (decls2, stmts2) = codegen(body, mlsRetValue)(using decls, stmts)
+      val stmtsWithReturn = stmts2 :+ Stmt.Return(Expr.Var(mlsRetValue))
+      val theDef = Def.FuncDef(mlsValType, name |> mapName, params.map(x => (x |> mapName, mlsValType)), Stmt.Block(decls2, stmtsWithReturn))
+      val decl = Decl.FuncDecl(mlsValType, name |> mapName, params.map(x => mlsValType))
+      (theDef, decl)
+
+  private def codegenTopNode(node: Node)(using ctx: Ctx): (Def, Decl) =
+    val decls = Ls(mlsRetValueDecl)
+    val stmts = Ls.empty[Stmt]
+    val (decls2, stmts2) = codegen(node, mlsRetValue)(using decls, stmts)
+    val stmtsWithReturn = stmts2 :+ Stmt.Return(Expr.Var(mlsRetValue))
+    val theDef = Def.FuncDef(mlsValType, mlsMainName, Ls(), Stmt.Block(decls2, stmtsWithReturn))
+    val decl = Decl.FuncDecl(mlsValType, mlsMainName, Ls())
+    (theDef, decl)
+
+  private def sortClasses(prog: Program): Ls[ClassInfo] =
+    var depgraph = prog.classes.map(x => (x.name, x.parents)).toMap
+    var degree = depgraph.view.mapValues(_.size).toMap
+    def removeNode(node: Str) =
+      degree -= node
+      depgraph -= node
+      depgraph = depgraph.view.mapValues(_.filter(_ != node)).toMap
+      degree = depgraph.view.mapValues(_.size).toMap
+    var sorted = Ls.empty[ClassInfo]
+    var work = degree.filter(_._2 == 0).keys.toSet
+    while work.nonEmpty do
+      val node = work.head
+      work -= node
+      sorted = sorted :+ prog.classes.find(_.name == node).get
```

Comment:
This is terribly inefficient. Just use a `mutable.Buffer` (such as a `mutable.ListBuffer`) assigned to a `val` here, instead of a `List` assigned to a `var`.